src/main/kotlin/org/rust/lang/core/resolve/indexes/RsImplIndex.kt

```diff
@@ -56,6 +72,14 @@ class RsImplIndex : AbstractStubIndex<RustTypeFingerprint, RsImplItem>() {
                 // struct S; impl<S: Tr1> Tr2 for S {}
                 ty != null && ty.javaClass == target.javaClass && ty.canUnifyWith(target, project)
             }
+            val traitDeref = collection.find { it.traitRef?.resolveToTrait?.langAttribute == "deref" }
```

Comment:
Yeah, the `langItems` is the right approach here! 

Let's perhaps create a `src/main/kotlin/org/rust/lang/core/resolve/LangItems.kt` file and move the lang-items related logic there? We have a ton of similar use case already, and it's high time we start consolidate it. 

For know, I think the file needs just one `fun isDerefTrait(trait: RsTraitItem): Boolean` one-liner.